pkg/standalone/uninstall.go

```diff
@@ -10,47 +10,60 @@ import (
 
 // Uninstall deletes all installed containers
 func Uninstall(uninstallAll bool, dockerNetwork string) error {
-	var errs []error
+	var containerErrs []error
 
-	_, err := utils.RunCmdAndWait(
-		"docker", "rm",
-		"--force",
-		utils.CreateContainerName(DaprPlacementContainerName, dockerNetwork))
-
-	if err != nil {
-		errs = append(errs, fmt.Errorf("could not remove %s container: %s", DaprPlacementContainerName, err))
-	}
-
-	_, err = utils.RunCmdAndWait(
-		"docker", "rmi",
-		"--force",
-		daprDockerImageName)
+	dockerInstalled := utils.IsDockerInstalled()
+	if !dockerInstalled {
+		containerErrs = append(
+			containerErrs,
+			fmt.Errorf("could not connect to Docker. Docker may not be installed or running"))
+	} else {
+		_, err := utils.RunCmdAndWait(
+			"docker", "rm",
+			"--force",
+			utils.CreateContainerName(DaprPlacementContainerName, dockerNetwork))
 
-	if err != nil {
-		errs = append(errs, fmt.Errorf("could not remove %s container: %s", daprDockerImageName, err))
-	}
+		if err != nil {
+			containerErrs = append(
+				containerErrs,
+				fmt.Errorf("could not remove %s container: %s", DaprPlacementContainerName, err))
+		}
 
-	if uninstallAll {
 		_, err = utils.RunCmdAndWait(
-			"docker", "rm",
+			"docker", "rmi",
 			"--force",
-			utils.CreateContainerName(DaprRedisContainerName, dockerNetwork))
+			daprDockerImageName)
+
 		if err != nil {
-			errs = append(errs, fmt.Errorf("could not remove %s container: %s", DaprRedisContainerName, err))
+			containerErrs = append(
+				containerErrs,
+				fmt.Errorf("could not remove %s container: %s", daprDockerImageName, err))
```

Comment:
nit: should this message say image instead of container

Reply:
Good catch 👍 I've made the change and also moved the removal of containers into a method so that it's more readable.